src/agent/transform-headers-agent.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-underscore-dangle */
+const http = require('http');
+const WrappedAgent = require('./wrapped-agent');
+
+const { _storeHeader } = http.OutgoingMessage.prototype;
+
+class TransformHeadersAgent extends WrappedAgent {
+    transformRequest(request) {
+        const headers = {};
+        const hasConnection = request.hasHeader('connection');
+        const hasContentLength = request.hasHeader('content-length');
+        const hasTransferEncoding = request.hasHeader('transfer-encoding');
+        const hasTrailer = request.hasHeader('trailer');
+        const keys = request.getHeaderNames();
+
+        for (const key of keys) {
+            headers[key] = request.getHeader(key);
+            request.removeHeader(key);
+        }
+
+        if (!hasConnection) {
+            const shouldSendKeepAlive = request.shouldKeepAlive && (hasContentLength || request.useChunkedEncodingByDefault || request.agent);
+            if (shouldSendKeepAlive) {
+                headers[this.transformHeader('connection')] = 'keep-alive';
```

Comment:
Interested to hear why not just use `Connection` in those cases where you're transforming a constant.

Reply:
https://github.com/apify/got-scraping/pull/26#discussion_r681794499